sdk/core/azure-json/src/main/java/com/azure/json/JsonReader.java

```diff
@@ -592,7 +592,7 @@ public final Object readUntyped() throws IOException {
 
     private Object readUntypedHelper(int depth) throws IOException {
         // Keep track of array and object nested depth to prevent a StackOverflowError from occurring.
-        if (depth >= 1000) {
+        if (depth >= 999) {
```

Comment:
hacky fix for [test failure]([fixes](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2744328&view=logs&j=a4d29859-3a3e-58e9-1e57-ab36931b8c3d&t=380f79dc-1fa7-5f4f-96cd-1232e7eff958))

```JacksonJsonReaderContractTests>JsonReaderContractTests.readUntypedPreventsStackOverflow:559 Unexpected exception type thrown, expected: <java.lang.IllegalStateException> but was: <com.fasterxml.jackson.core.exc.StreamConstraintsException>```

Jackson now throws `com.fasterxml.jackson.core.exc.StreamConstraintsException: Depth (1001) exceeds the maximum allowed nesting depth (1000)`, so we have 1001  nested calls here.

Reply:
I was looking into this further to see if we can infer the existing stack depth from the underlying implementation, it looks possible in Jackson 2.15 as deserialization context now tracks depth but I wasn't able to find anything in GSON.

This likely should get updated to track overall stack depth as we could already be 10-20 layers deep but for now this should be more than sufficient.